### PR TITLE
Add valid_url? helper

### LINKED ISSUE #158: Helper function to check validity of a URL
Since we have all the regex already, it would be nice to have a helper function to check if a URL is a valid video URL. Something like `VideoInfo.valid_url?(url)` that checks against all of the providers' URL regex.

### PR DESCRIPTION
**Pull request by markets**
Closes #158

Add new helper `valid_url?` to check if a url is valid in some of the enabled providers:

```ruby
> VideoInfo.valid_url?('http://www.youtube.com/watch?v=AT_5xOGh6Ko')
=> true
> VideoInfo.valid_url?('http://vimeo.com/898029')
=> true
> VideoInfo.valid_url?('http://www.example.com/video/12345')
=> false
```

### CODE REVIEW

File: lib/video_info.rb [hunk 1]
```diff
@@ -81,6 +81,13 @@ def self.disabled_provider?(provider)
     disable_providers.map(&:downcase).include?(provider.downcase)
   end
 
+  def self.valid_url?(url)
+    PROVIDERS.select { |p| !disabled_provider?(p) }.any? do |provider|
```
Comment (thibaudgg) [inline lib/video_info.rb:85]: I would add a separate methods to return the enabled_providers.


``` ruby
def self.enabled_providers
  PROVIDERS
    .reject { |p| disabled_provider?(p) }
    .map { |p| Providers.const_get(p) }
end
```

File: lib/video_info.rb [hunk 2]
```diff
@@ -81,6 +81,13 @@ def self.disabled_provider?(provider)
...
+      provider_const = Providers.const_get(provider)
+      return true if provider_const.usable?(url)
+    end
+  end
```
Comment (thibaudgg) [inline lib/video_info.rb:89]: With the new `enabled_providers` method that would just be:

```suggestion
  def self.valid_url?(url)
    enabled_providers any? { |p| p.usable?(url) }
  end
```

**Comment by markets** [comment]
@thibaudgg updated with your suggestions + mention in the README.

**Comment by thibaudgg** [comment]
@markets thanks for the update. 👍

Yes, some work would be required to fix the build and update it to support the latest Ruby version, but I don't have much time for it now. Please feel free to have a look and submit a PR to improve it.

**Comment by thibaudgg** [comment]
3.1.0 released.

**Comment by miguelpeniche** [comment]
Hi there @thibaudgg. There is a minor bug: with the Vimeo user page `VideoInfo.valid_url? "https://vimeo.com/user123456789"` is returning `true`.

**Pull request merged**